hydrolib/core/io/inifield/models.py

```diff
@@ -0,0 +1,244 @@
+
+
+class IniFieldGeneral(INIGeneral):
+    """The initial field file's `[General]` section with file meta data."""
+
+    class Comments(INIBasedModel.Comments):
+        fileversion: Optional[str] = Field(
+            "File version. Do not edit this.", alias="fileVersion"
+        )
+        filetype: Optional[str] = Field(
+            "File type. Should be 'iniField'. Do not edit this.",
+            alias="fileType",
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["General"] = "General"
+    fileversion: str = Field("2.00", alias="fileVersion")
+    filetype: Literal["iniField"] = Field("iniField", alias="fileType")
+
+
+class AbstractSpatialField(INIBasedModel, ABC):
+    """
+    Abstract base class for `[Initial]` and `[Parameter]` block data in
+    inifield files.
+
+    Defines all common fields. Used via subclasses InitialField and ParameterField.
+    """
+
+    class Comments(INIBasedModel.Comments):
+        quantity: Optional[str] = Field(
+            "Name of the quantity. See UM Table D.2.", alias="quantity"
+        )
+        datafile: Optional[str] = Field(
+            "Name of file containing field data values.", alias="dataFile"
+        )
+        datafiletype: Optional[str] = Field("Type of dataFile.", alias="dataFileType")
+        interpolationmethod: Optional[str] = Field(
+            "Type of (spatial) interpolation.", alias="interpolationmethod"
+        )
+        operand: Optional[str] = Field(
+            "How this data is combined with previous data for the same quantity (if any).",
+            alias="operand",
+        )
+        averagingtype: Optional[str] = Field(
+            "Type of averaging, if interpolationMethod=averaging .",
+            alias="averagingtype",
+        )
+        averagingrelsize: Optional[str] = Field(
+            "Relative search cell size for averaging.", alias="averagingrelsize"
+        )
+        averagingnummin: Optional[str] = Field(
+            "Minimum number of points in averaging. Must be ≥ 1.",
+            alias="averagingnummin",
+        )
+        averagingpercentile: Optional[str] = Field(
+            "Percentile value for which data values to include in averaging. 0.0 means off.",
+            alias="averagingpercentile",
+        )
+        extrapolationmethod: Optional[str] = Field(
+            "Option for (spatial) extrapolation.", alias="extrapolationmethod"
+        )
+        locationtype: Optional[str] = Field(
+            "Target location of interpolation.", alias="locationtype"
+        )
+        value: Optional[str] = Field(
+            "Only for dataFileType=polygon. The constant value to be set inside for all model points inside the polygon."
+        )
+
+    comments: Comments = Comments()
+
+    quantity: str = Field(alias="quantity")
+    datafile: Path = Field(alias="dataFile")
+    datafiletype: DataFileType = Field(alias="dataFileType")
+    interpolationmethod: Optional[InterpolationMethod] = Field(
```

Comment:
Should interpolationMethod really be optional? I do not see it in the manual.

Reply:
It is optional, because when `dataFileType=1dField`, no interpolation method is necessary